### PR TITLE
JDK-8061729 : Update java/net tests to eliminate dependency on sun.net.www.MessageHeader and some other internal APIs

### PR DESCRIPTION
**Pull request by mahendrachhipa**
There are some regression tests depending on sun.net.www.MessageHeader, the internal API dependency should be removed. Some of other internal API dependancies are removed in following issues :
JDK-8273142
JDK-8268464
JDK-8268133

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8061729](https://bugs.openjdk.java.net/browse/JDK-8061729): Update java/net tests to eliminate dependency on sun.net.www.MessageHeader and some other internal APIs

### Reviewers
 * [Daniel Fuchs](https://openjdk.java.net/census#dfuchs) (@dfuch - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5937/head:pull/5937` \
`$ git checkout pull/5937`

Update a local copy of the PR: \
`$ git checkout pull/5937` \
`$ git pull https://git.openjdk.java.net/jdk pull/5937/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5937`

View PR using the GUI difftool: \
`$ git pr show -t 5937`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5937.diff">https://git.openjdk.java.net/jdk/pull/5937.diff</a>

</details>

### CODE REVIEW

**Comment by dfuch** [comment]
It would be good to have a separate test to test the behavior of the new library `HttpHeaderParser` class.
As an example, here [1] is the test we have written for the HttpClient Http1HeaderParser. You will see that it defines a numbers of test cases (response headers) - both positive and negative - and verifies that `sun.net.www.MessageHeader` and `Http1HeaderParser` give the same results in the positive case. It would be good to have something similar for `HttpHeaderParser` as it would give more confidence that the parsing is correct. You would need of course to add in your test new test cases that contain a request line - the Http1HeaderParser only parses status line since it's only used on the client side.

[1] https://github.com/openjdk/jdk/blob/master/test/jdk/java/net/httpclient/whitebox/java.net.http/jdk/internal/net/http/Http1HeaderParserTest.java

File: test/lib/jdk/test/lib/net/HttpHeaderParser.java [hunk 1]
```diff
@@ -80,4 +109,278 @@ public HttpHeaderParser(InputStream is) throws IOException {
...
+     *          if the input buffer's current position is not smaller
+     *          than its limit
+     */
+    private char get(InputStream input) throws IOException {
+        return (char)(input.read() & 0xFF);
+    }
+
+    private void readResumeStatusLine(InputStream input) throws IOException {
+        char c = 0;
+        while (input.available() > 0 && (c = get(input)) != CR) {
```
Comment (dfuch) [inline test/lib/jdk/test/lib/net/HttpHeaderParser.java:175]: It's not a too good idea to depend on input.available() - the spec leaves too much room for odd behaviors.
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/io/InputStream.html#available()
An implementation may decide to return 0 always - as it is the amount of bytes that is guaranteed to be readable without blocking (and an implementation may not know how many bytes can be read without blocking).

**Comment by mahendrachhipa** [comment]
Mach5 results https://mach5.us.oracle.com/mdash/jobs/mahendrachhipa-jdk-20220216-1342-29188746

**Review by dfuch — APPROVED** [review]

**Pull request closed without merging**